dbt/include/databricks/macros/adapters.sql

```diff
@@ -105,16 +115,234 @@
+    {% set column = column_dict[column_name] %}
+    {% set constraints = get_column_constraints(column)  %}
+
+    {% set statements = get_constraints_sql(relation, constraints, model, column) %}
+    {% for stmt in statements %}
+      {% call statement() %}
+        {{ stmt }}
+      {% endcall %}
+    {% endfor %}
+
+  {% endfor %}
+
+{% endmacro %}
+
+{% macro get_constraints_sql(relation, constraints, model, column) %}
+  {% set statements = [] %}
+  {% for constraint in constraints %}
+    {% if constraint %}
+      {% set constraint_statements = get_constraint_sql(relation, constraint, model, column) %}
+      {% for statement in constraint_statements %}
+        {% if statement %}
+          {% do statements.append(statement) %}
+        {% endif %}
+      {% endfor %}  
+    {% endif %}
+  {% endfor %}
+
+  {{ return(statements) }}
+{% endmacro %}
+
+{% macro get_constraint_sql(relation, constraint, model, column={}) %}
+{{ exceptions.warn("get_constraint_sql") }}
+  {% set statements = [] %}
+  {% set type = constraint.get("type", "") %}
+
+  {% if type == 'check' %}
+    {% set expression = constraint.get("expression", "") %}
+    {% if not expression %}
+      {{ exceptions.raise_compiler_error('Invalid check constraint expression: ' ~ expression) }}
+    {% endif %}
+
+    {% set name = constraint.get("name") %}
+    {% if not name and local_md5 %}
+      {%- set name = local_md5 (column.get("name", "") ~ ";" ~ expression ~ ";") -%}
+    {% endif %}
+    {% set stmt = "alter table " ~ relation ~ " add constraint " ~ name ~ " check (" ~ expression ~ ");" %}
+    {% do statements.append(stmt) %}
+  {% elif type == 'not_null' %}
+    {% set column_names = constraint.get("columns", []) %}
+    {% if column and not column_names %}
+      {% set column_names = [column['name']] %}
+    {% endif %}
+    {% for column_name in column_names %}
+      {% set column = model.get('columns', {}).get(column_name) %}
+      {% if column %}
+        {% set quoted_name = adapter.quote(column['name']) if column['quote'] else column['name'] %}
+        {% set stmt = "alter table " ~ relation ~ " change column " ~ quoted_name ~ " set not null " ~ (constraint.expression or "") ~ ";" %}
+        {% do statements.append(stmt) %}
+      {% else %}
+        {{ exceptions.warn('not_null constraint on invalid column: ' ~ column_name) }}
+      {% endif %}
+    {% endfor %}
+  {% elif type == 'primary_key' %}
+    {% if constraint.get('warn_unenforced') %}
+      {{ exceptions.warn("unenforced constraint type: " ~ type)}}
+    {% endif %}
+    {% set column_names = constraint.get("columns", []) %}
+    {% if column and not column_names %}
+      {% set column_names = [column['name']] %}
+    {% endif %}
+    {% set quoted_names = [] %}
+    {% for column_name in column_names %}
+      {% set column = model.get('columns', {}).get(column_name) %}
+      {% if not column %}
+        {{ exceptions.warn('Invalid primary key column: ' ~ column_name) }}
```

Comment:
isn't column_name always empty here?

Reply:
column would be empty, but this is the case where the user specifies an invalid column name in the columns property of the constraint